restapi/src/main/java/io/stargate/web/resources/v2/RowsResource.java

```diff
@@ -380,8 +382,14 @@ public Response createRow(
           AuthenticatedDB authenticatedDB =
               db.getRestDataStoreForToken(token, getAllHeaders(request));
 
-          @SuppressWarnings("unchecked")
-          Map<String, Object> requestBody = mapper.readValue(payload, Map.class);
+          Map<String, Object> requestBody;
+          try {
+            requestBody =
+                mapper.readValue(payload, new TypeReference<HashMap<String, Object>>() {});
+          } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(
```

Comment:
As per my comment on the issue, it might make sense to further separate out exception types; `JsonParseException` subtype would be for low-level decoding/parsing.
Probably does not make much difference here since there isn't much chance of data-binding (json tokens -> Java objects) problems.